lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -888,6 +888,14 @@ LogicalResult NEQPrimOp::canonicalize(NEQPrimOp op, PatternRewriter &rewriter) {
 // Unary Operators
 //===----------------------------------------------------------------------===//
 
+OpFoldResult SizeOfIntrinsicOp::fold(llvm::ArrayRef<mlir::Attribute>) {
+  auto base = getInput().getType().dyn_cast<FIRRTLBaseType>();
```

Comment:
nit: i think either `base` should be checked for null, or else this can be a `cast`.